finality-aleph/src/party/manager/mod.rs

```diff
@@ -465,13 +465,13 @@ where
     }
 
     async fn node_idx(&self, authorities: &[AuthorityId]) -> Option<NodeIndex> {
-        let our_consensus_keys: HashSet<_> = self
-            .keystore
-            .keys(KEY_TYPE)
-            .await
-            .unwrap()
-            .into_iter()
-            .collect();
+        let our_consensus_keys: HashSet<_> = match self.keystore.keys(KEY_TYPE).await {
+            Ok(keys) => keys.into_iter().collect(),
+            Err(e) => {
+                warn!(target: "aleph-data-store", "No validator keys present on machine: {}", e);
+                return None;
+            }
+        };
```

Comment:
This code is used to run validators and nonvalidators now, the latter technically might be missing keys in which case this panicked. I don't think that was the case in practice (it seems nodes generate some keys even if they are not validators?), but I dunno.

The trigger for this PR was flaky (not always failing) tests which failed here – at best they will stop being flaky, but since they were I don't expect this to actually help with them. It does make the logic clearer though – if we don't have keys for a session we just don't run in the session.

Reply:
The node does not generate keys when not asked to, so I'm still kind of unsure which case it is solving. 

The reason why I'm asking "why did it used to work and now doesn't" is because I feel like we are trying to treat symptoms here without understanding the root cause.

Reply:
In this case I doubt it'll even treat the symptoms then, since it only failed nondeterministically (so the node had to sometimes generate keys, so probably it was always told to generate keys, so it's likely to be a problem with something else). This PR is mostly trying to make the logic behind this function more consistent.

Reply:
Oh, and if it's the case that you can run a node without keys, then this makes it actually work.

Reply:
I got lost here: are you claiming that currently (without this PR) running without keys crashes aleph-node? If so, in what circumstances?

Reply:
I don't actually know – I know this sometimes fails in our pipelines with a "file not found" error (see the slack discussion where you found this PR). Maybe this file should always exists, even when empty? Then the node usually wouldn't crash, even if it doesn't generate keys, but apparently crashes in some async scenario where it fails to create this file in time/at all?

Or maybe better – since you say that keys are not always created, how do I run a node without keys, to check whether it'll crash?

Reply:
So I just checked it and it seems to not crash (but there is some probability I made a mistake).

You can try adding `rm -r $BASE_PATH/$account_id/chains/a0dnet1/keystore/` below this line https://github.com/Cardinal-Cryptography/aleph-node/blob/66d20a2aa14af2c7c11892274e3a79db1536b36f/scripts/run_nodes.sh#L63

And then run `./scripts/run_nodes.sh -v 4 -n 1` and then the non-validaotr (node number 4) will not have keys.

Reply:
I'm pretty sure that the `unwrap` that we are trying to "cure" here is not the problem, i.e., unwrap failing does not mean we don't have the keys. It's a different error.

Reply:
Yup, that's probably true then, I also tried this test on my machine and it also doesn't fail. I still think this PR makes the function behave more consistently, and that's the main reason why I wanted it in the first place.

As for the actual problem, I have no idea what's wrong there, especially since the "file not found" error kinda should've turned up in the test you proposed just now, no?

Reply:
Ok, I think it's good to leave the code as is, but I think that the log should say something like `Error when accessing the keystore` rather than what it says now. Ideally we should check in substrate when does it return Error